cmd/maya-exporter/app/collector/metrics.go

```diff
@@ -0,0 +1,224 @@
+	Conn    net.Conn
+	Metrics Metrics
+}
+
+// JivaStatsExporter implements the prometheus.Collector interface. It exposes
+// the metrics of a OpenEBS (Jiva) volume.
+type JivaStatsExporter struct {
+	VolumeControllerURL string
+	Metrics             Metrics
+}
+
+// A gauge is a metric that represents a single numerical value that can
+// arbitrarily go up and down.
+
+// Gauges are typically used for measured values like temperatures or current
+// memory usage, but also "counts" that can go up and down, like the number of
+// running goroutines.
+
+// GaugeOpts is the alias for Opts, which is used to create diffent type of
+// metrics.
+
+// GaugeVec is a Collector that bundles a set of Gauges that all share the same
+// Desc, but have different values for their variable labels. This is used if
+// you want to count the same thing partitioned by various dimensions
+// (e.g. number of operations queued, partitioned by user and operation
+// type). Create instances with NewGaugeVec.
+
+// CounterVec is a Collector that bundles a set of Counters that all share the
+// same Desc, but have different values for their variable labels. This is used
+// if you want to count the same thing partitioned by various dimensions
+// (e.g. number of HTTP requests, partitioned by response code and
+// method). Create instances with NewCounterVec.
+//
+
+// Metrics keeps all the volume related stats values into the respective fields.
+type Metrics struct {
+	actualUsed             prometheus.Gauge
+	logicalSize            prometheus.Gauge
+	sectorSize             prometheus.Gauge
+	readIOPS               prometheus.Gauge
+	readTimePS             prometheus.Gauge
+	readBlockCountPS       prometheus.Gauge
+	writeIOPS              prometheus.Gauge
+	writeTimePS            prometheus.Gauge
+	writeBlockCountPS      prometheus.Gauge
+	readLatency            prometheus.Gauge
+	writeLatency           prometheus.Gauge
+	avgReadBlockCountPS    prometheus.Gauge
+	avgWriteBlockCountPS   prometheus.Gauge
+	sizeOfVolume           prometheus.Gauge
+	volumeUpTime           *prometheus.GaugeVec
+	connectionRetryCounter *prometheus.CounterVec
+	connectionErrorCounter *prometheus.CounterVec
+}
+
+// MetricsDiff keep the difference of the read/write I/O's and
+// other volume statistics per second.
+type MetricsDiff struct {
+	readIOPS             float64
+	writeIOPS            float64
+	readBlockCountPS     float64
+	writeBlockCountPS    float64
+	readTimePS           float64
+	writeTimePS          float64
+	readLatency          float64
+	writeLatency         float64
+	avgReadBlockCountPS  float64
+	avgWriteBlockCountPS float64
+	size                 float64
+	sectorSize           float64
+	logicalSize          float64
+	actualSize           float64
+	uptime               float64
+}
+
+var (
+	// Declare the Metrics instance to reuse it in registration
+	// of exporter while instantiating JivaStatsExporter and
+	// CstorStatsExporter.
+	metrics = Metrics{
```

Comment:
I guess, this is the metrics i.e. gauge/counter etc registration.
Can this be a valid method of Metrics struct & be invoked from the functions that create new instances of CStorExporter & JivaExporter?

Reply:
looks good,